src/ctypes-foreign-threaded/foreign.mli

```diff
@@ -93,3 +104,88 @@ val funptr_opt :
 exception CallToExpiredClosure
 (** A closure passed to C was collected by the OCaml garbage collector before
     it was called. *)
+
+module type Funptr = sig
+  type fn
+  (** [fn] is the signature of the underlying ocaml function. *)
+
+  type t
+  (** Handle to an ocaml function that can be passed to C for use in callbacks.
+
+      Unfortunately it is not possible to track if a fuction pointer is still used in C,
+      so you must use the appropriate life cycle functions below for this.
+
+      See [t], [of_fun], [with_fun] and [free] *)
+
+  val t : t Ctypes.typ
+  (** ctype that can be used in type signatures of external functions *)
+
+  val t_opt : t option Ctypes.typ
+  (** Like [t] but optional. *)
+
+  val free : t -> unit
+  (** [free fptr] - Indicate that the [fptr] is no longer needed.
+
+      Once [free] has been called any C calls to this [Dynamic_funptr.t] are
+      unsafe and may result in a segmentation fault. Only call [free] once the
```

Comment:
Drop "and may result in a segmentation fault"

Reply:
Done